backend/gateway/config.go

```diff
@@ -105,12 +106,19 @@ func parseFile(path string, pb proto.Message, template bool) error {
 		}
 	}
 
-	// Interpolate environment variables.
-	contents = []byte(os.ExpandEnv(string(contents)))
+	// Interpolate environment variables
+	expandedData := os.ExpandEnv(string(contents))
+
+	contents = []byte(processTemplateToken(expandedData))
 
 	return parseYAML(contents, pb)
 }
 
+// Replace the Clutch-specific templating token with the Go $ template syntax
+func processTemplateToken(data string) string {
```

Comment:
background: we want to support [Go templating syntax](https://golang.org/pkg/text/template/#hdr-Variables) in `clutch-config.yaml`

`os.ExpandEnv` removes the `$` (and it's [not currently possible](https://github.com/golang/go/issues/43482) to escape the `$`), so we'll need to use our own templating token in the config yaml and then replace it with a `$`, _after_ `os.ExpandEnv` has been called.

as for the clutch templating token, I choose `%%` but it can be something else.
example
```
message: {{range %%v, %%k := .Bar}}{{%%k}}:{{%%v}}{{end}}
```

Reply:
great explanation! let's move it to the godoc.

also i think you will also need to choose a replacement token for the curly brackets... since we run go templating on the file itself at lyft (`-template` to the binary)

one other option, at least for `$` is to replace instances of `$$` with some token first, then do the environment variable expansion, then replace them back to single `$`. e.g. `replaceAll('$$', '@#@') -> os.ExpandEnv -> replaceAll('@#@', '$')`

Reply:
RE template in a template - thanks for calling that out! wasn't hitting this case https://github.com/lyft/clutch/blob/main/backend/gateway/config.go#L101.

Latest commit changes
1) $$ -> @#A -> $
2) [[ ]] -> {{ }}

Tested locally, added unit tests, and also a comment explaining the issue/solution